template/java8/entrypoint/src/main/java/com/openfaas/entrypoint/App.java

```diff
@@ -73,8 +74,9 @@ public void handle(HttpExchange t) throws IOException {
             //     System.out.println("Req header " + entry.getKey() + " " + entry.getValue());
             // }
 
-            IRequest req = new Request(requestBody, reqHeadersMap);
-
+            String queryRaw = t.getRequestURI().getRawQuery();
+            IRequest req = new Request(requestBody, reqHeadersMap,getQueryParameters(queryRaw),queryRaw);
```

Comment:
What if we moved the method "getQueryParameters" into the Request class, would that make sense?

Reply:
Yeah, we can do that. Here I am considering Request as a model and all injection is done by server. In this case , App is responsible to inject params in request model.
thoughts?